sci-libs/primegen/primegen-0.97.recipe

```diff
@@ -0,0 +1,78 @@
+SUMMARY="Fast C/C++ prime number generator library"
+DESCRIPTION="primegen is a small, fast library to generate prime numbers \
+in order. It generates the 50847534 primes up to 1000000000 in just 8 \
+seconds on a Pentium II-350; it prints them in decimal in just 35 seconds. \
+primegen can generate primes up to 1000000000000000, although it is not \
+optimized for primes past 32 bits. It uses the Sieve of Atkin instead of the \
+traditional Sieve of Eratosthenes."
+HOMEPAGE="https://cr.yp.to/primegen.html"
+COPYRIGHT="1999, D. J. Bernstein"
+LICENSE="Public Domain"
+REVISION="1"
+SOURCE_URI="https://cr.yp.to/primegen/primegen-$portVersion.tar.gz"
+CHECKSUM_SHA256="54285baf8eed9e421ff2220a2112d38cfb20c1ebef6014ef3f0004c22c95f40d"
+PATCHES="primegen-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	primegen = $portVersion
+	cmd:primegaps = $portVersion
+	cmd:primes = $portVersion
+	cmd:primespeed = $portVersion
+	lib:libprimegen
+	"
+REQUIRES="
+	haiku
+	"
+
+PROVIDES_devel="
+	primegen_devel = $portVersion
+	devel:libprimegen
+	"
+REQUIRES_devel="
+	primegen == $portVersion base
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	cmd:nroff
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+BUILD()
+{
+	make
+}
+
+INSTALL()
+{
+	install -d $binDir $docDir $manDir/man1 $manDir/man3
+	install -d $includeDir/primegen $developLibDir $libDir
+	install -t $binDir primegaps primes primespeed
+	install -t $docDir BLURB README
+	install -t $manDir/man1 primegaps.1 primes.1 primespeed.1
+	install -t $manDir/man3 primegen.3
+	install -t $includeDir/primegen primegen.h uint32.h uint64.h
+	install -T primegen.a $developLibDir/libprimegen.a
+	install -T primegen.so $libDir/libprimegen.so
+	prepareInstalledDevelLib libprimegen
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir \
+		$manDir/man3
+}
+
+TEST()
+{
+	# 32-bit test
+	[ "$(./primes 1 1000000 | md5sum | cut -d " " -f1)" == "c13929ee9d2aea8f83aa076236079e94" ]
+	# all tests passed
+	exit 0
```

Comment:
Oops, you forgot to remove the `exit 0`.
The exit code returned by `TEST()` is the exit code of the *last* command in that function, so, since we have removed the ` && exit 0` at the end of the `| ... ]` test, we should not run any command avec the test.

Oops, you also used `==` in the test, while you should use `=`. Both work... with ``bash``, but with ``sh`` the only accepted syntax for testing if 2 strings are identical is `=`. Using bash-epecific syntax is not recommended when something can be done (with no extra cost) using sh-syntax.

BTW, the c13929ee9d2aea8f83aa076236079e94 string does not need to be quoted, since it does not contain any special caracter. Well, keeping the quotes does not harm, but we can safely drop them.
(On the left side of the test, the quotes around the `$(...)` are fine and it would not be safe to remove them.)

So here what you can put in TEST to fix these tiny issues:
```
TEST()
{
	[ "$(./primes 1 1000000 | md5sum | cut -d " " -f 1)" = c13929ee9d2aea8f83aa076236079e94 ]
}
```
Regarding the `cut` command, the `-d` and `-f` options both accept a space before their values, but it's nice to either have no space at all for each of them, or a space each of them. Well, that's once again just style ;-)
BTW, you could also use `cut -d\  -f1` but in that case you really need two spaces after the backslash.

I hope you're not too angry with me ;-) because your first and second definitions of TEST were both correct, and then I suggested a style change...
So sorry again for my picky requests and thanks for your excellent work!

Reply:
Cool, you squashed you commits and chose a nice commit message.

After you fix TEST, remember that you can use
```
git commit --amend primegen-0.97.recipe
```
so that you won't need to redo a squash.

Reply:
Yup! Amended :)